connect/src/spirc.rs

```diff
@@ -693,9 +794,26 @@ impl SpircTask {
     }
 
     fn load_track(&mut self, play: bool) {
-        let index = self.state.get_playing_track_index();
+        let mut index = self.state.get_playing_track_index();
         let track = {
-            let gid = self.state.get_track()[index as usize].get_gid();
+            // let gid = self.state.get_track()[index as usize].get_gid();
+            let track_ref = self.state.get_track()[index as usize].to_owned();
+            let mut gid = track_ref.get_gid();
+            if gid.len() != 16 {
+                let track_len = self.state.get_track().len() as u32;
+                if index < track_len - 1 {
+                    index = 0;
+                } else {
+                    index = index + 1;
```

Comment:
This is fine if the odd song has a malformed gid, but 2 consecutive songs with a bad gid will cause a crash with this code i believe? given the relative rarity of malformed gids, might i suggest we run a check against the next tracks gid as well, and loop this check until a valid gid is found, before continuing execution? Might seem like overkill, but it's one less wierd bug that we'll have to solve at a later date.

Reply:
Done!